extractor.go

```diff
@@ -35,15 +35,11 @@ func AuthHeaderTokenExtractor(r *http.Request) (string, error) {
 func CookieTokenExtractor(cookieName string) TokenExtractor {
 	return func(r *http.Request) (string, error) {
 		cookie, err := r.Cookie(cookieName)
-		if err != nil {
-			return "", err
+		if err == http.ErrNoCookie {
```

Comment:
Should it be `err != nil` then?

Reply:
The only error that the `Cookie()` func returns is `ErrNoCookie` tho:

```
func (r *Request) Cookie(name string) (*Cookie, error) {
	for _, c := range readCookies(r.Header, name) {
		return c, nil
	}
	return nil, ErrNoCookie
}
```

I could go either way but then if we do `err != nil` it should maybe look like:

```go
if err != nil {
  if err == http.ErrNoCookie {
    return "", nil
  }
  return "", err
}
```

Wdyt @Widcket ?

Reply:
It's likely that `Cookie()` may not need to produce another error in the foreseeable future, so it's probably fine either way. Just me nitpicking tho :)